src/EFTools/EntityDesign/VisualStudio/ModelWizard/ModelObjectItemWizard.cs

```diff
@@ -298,6 +302,20 @@ private static void PopluateReplacementDictionary(Project project, Dictionary<st
         [SuppressMessage("Microsoft.Maintainability", "CA1506:AvoidExcessiveClassCoupling")]
         public void RunFinished()
         {
+#if VS14ORNEWER
+            TelemetryService.DefaultSession.PostEvent(
+                new TelemetryEvent("vs/entityframework/entitydatamodelwizard")
+                {
+                    Properties =
+                    {
+                        {
+                            "VS.EntityFramework.EntityDataModelWizard.GenerationOption",
+                            _modelBuilderSettings.GenerationOption
```

Comment:
How does this get serialized? An int, an enum value or the or the ToString() of the  `ModelGenerationOption` member?

Are there restrictions on what it can be? 

It seems that the string could be more future-proof if e.g., if we implement https://github.com/aspnet/EntityFramework6/issues/302 it could keep working the same for the existing options but new options could be added with a unique string.